spec/lobstersbot/title_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'spec_helper'
+require 'uri'
+
+RSpec.describe Lobstersbot::Title do
+  subject { klass.new }
+
+  let(:klass) do
+    Class.new do
+      include Lobstersbot::PluggableConnection
+      include Lobstersbot::Title
+    end
+  end
+
+  describe '#fetch' do
+    it 'fetches titles' do
+      title = Lobstersbot::Title.fetch(URI("https://lobste.rs/")) do |r|
+        Lobstersbot::Title.parse_title(r)
+      end
+      expect(title).to eq("Lobsters")
+    end
+
+    it 'avoids redirect loops' do
+      title = Lobstersbot::Title.fetch(
+        URI("https://demo.cyotek.com/features/redirectlooptest.php")) do |r|
+        Lobstersbot::Title.parse_title(r)
+      end
+      expect(title).to be_nil
+    end
+  end
+
+  # XXX: Should this be disabled? Apparently it's bad style
```

Comment:
Yes, it also triggers `SubjectStub` unfortunately. I do think it would be nice to have a `Lobstersbot::StubConnection` where the messages to `privmsg` etc. are output to an array as you mentioned, but I can fix that with a future PR.

Reply:
Until we have a fix for the limitations imposed by the current framework, you could add an `attr_acessor` to the mock class you create and pass the double in that way.